rackcity/views/asset_views.py

```diff
@@ -283,14 +289,22 @@ def asset_modify(request):
     )
     if failure_response:
         return failure_response
-
+    create_new_asset_cp = False
     if change_plan:
         failure_response = get_cp_already_executed_response(change_plan)
         if failure_response:
             return failure_response
-        del data["id"]
-        existing_asset = get_or_create_asset_cp(id, change_plan)
-        if existing_asset is None:
+        # del data["id"]
+
+        create_new_asset_cp = not AssetCP.objects.filter(
+            id=id, change_plan=change_plan.id
+        ).exists()
+
+        if not create_new_asset_cp:
+            existing_asset =  AssetCP.objects.get(
+            id=id, change_plan=change_plan.id
+        )
+        if not does_asset_exist(id, change_plan):
```

Comment:
I like the new method, but a little confused about this logic? worth talking about later